#DPATCHLEVEL=1

2004-10-26  Guillem Jover  <guillem@hadrons.org>

	* i386/i386/fpe.b: Remove binary without source.
	* i386/i386/fpe.b_elf: Likewise.
	* i386/i386/fpe_linkage.c: Disable the code if [! FPE].
	* i386/bogus/fpe.h: Add comment about not having an fpe implementation.
	* i386/Makefrag (objfiles): Do not add fpe.o.
	Remove targets to generate fpe.o.


diff -Naur gnumach-20040915.orig/i386/bogus/fpe.h gnumach-20040915/i386/bogus/fpe.h
--- gnumach-20040915.orig/i386/bogus/fpe.h	2001-06-25 18:28:26.000000000 +0200
+++ gnumach-20040915/i386/bogus/fpe.h	2004-10-26 11:27:49.000000000 +0200
@@ -1 +1,2 @@
+/* We do not have a floating point implementation.  */
 #define FPE 0
diff -Naur gnumach-20040915.orig/i386/i386/fpe_linkage.c gnumach-20040915/i386/i386/fpe_linkage.c
--- gnumach-20040915.orig/i386/i386/fpe_linkage.c	1997-02-25 22:27:09.000000000 +0100
+++ gnumach-20040915/i386/i386/fpe_linkage.c	2004-10-26 11:28:29.000000000 +0200
@@ -30,6 +30,8 @@
 
 #include <fpe.h>
 
+#if FPE
+
 #include <cpus.h>
 
 #include <mach/std_types.h>
@@ -357,3 +359,5 @@
 	}
 	exception(exc, code, subcode);
 }
+#endif /* FPE.  */
+
diff -Naur gnumach-20040915.orig/i386/Makefrag gnumach-20040915/i386/Makefrag
--- gnumach-20040915.orig/i386/Makefrag	2000-11-26 15:33:20.000000000 +0100
+++ gnumach-20040915/i386/Makefrag	2004-10-26 11:27:49.000000000 +0200
@@ -40,9 +40,6 @@
 objfiles += busses.o cirbuf.o
 vpath busses.c $(srcdir)/chips
 
-# FPE emulation
-objfiles += fpe.o
-
 # Mig-generated
 objfiles += mach_i386_server.o
 
@@ -132,11 +129,3 @@
 		rpc.h syscall_sw.h \
 		thread_status.h trap.h vm_param.h \
 		vm_types.h)
-
-
-# Cheat, cheat, cheat.
-fpe.o: fpe.b_elf
-	uudecode $<
-vpath fpe.b_elf $(sysdep)/i386
-
-fpe.d:; touch $@
